packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -2222,34 +2224,116 @@ function flushPassiveEffectsImpl() {
+    //
+    // Note: This currently assumes there are no passive effects on the root fiber
+    // because the root is not part of its own effect list.
+    // This could change in the future.
+    let effect = root.current.firstEffect;
+    let effectWithErrorDuringUnmount = null;
+    while (effect !== null) {
+      if (__DEV__) {
+        setCurrentDebugFiberInDEV(effect);
+        invokeGuardedCallback(
+          null,
+          commitPassiveHookUnmountEffects,
+          null,
+          effect,
+        );
+        if (hasCaughtError()) {
+          effectWithErrorDuringUnmount = effect;
+          invariant(effect !== null, 'Should be working on an effect.');
+          const error = clearCaughtError();
+          captureCommitPhaseError(effect, error);
+        }
+        resetCurrentDebugFiberInDEV();
+      } else {
+        try {
+          commitPassiveHookUnmountEffects(effect);
+        } catch (error) {
+          effectWithErrorDuringUnmount = effect;
+          invariant(effect !== null, 'Should be working on an effect.');
+          captureCommitPhaseError(effect, error);
+        }
       }
-      resetCurrentDebugFiberInDEV();
-    } else {
-      try {
-        commitPassiveHookEffects(effect);
-      } catch (error) {
-        invariant(effect !== null, 'Should be working on an effect.');
-        captureCommitPhaseError(effect, error);
+      effect = effect.nextEffect;
+    }
+
+    // Second pass: Create new passive effects.
+    //
+    // Note: This currently assumes there are no passive effects on the root fiber
+    // because the root is not part of its own effect list.
+    // This could change in the future.
+    effect = root.current.firstEffect;
+    while (effect !== null) {
+      // Don't run create effects for a Fiber that errored during destroy.
+      // This check is in place to match previous behavior.
+      // TODO: Rethink whether we want to carry this behavior forward.
+      if (effectWithErrorDuringUnmount !== effect) {
+        if (__DEV__) {
+          setCurrentDebugFiberInDEV(effect);
+          invokeGuardedCallback(
+            null,
+            commitPassiveHookMountEffects,
+            null,
+            effect,
+          );
+          if (hasCaughtError()) {
+            invariant(effect !== null, 'Should be working on an effect.');
+            const error = clearCaughtError();
+            captureCommitPhaseError(effect, error);
+          }
+          resetCurrentDebugFiberInDEV();
+        } else {
+          try {
+            commitPassiveHookMountEffects(effect);
+          } catch (error) {
+            invariant(effect !== null, 'Should be working on an effect.');
+            captureCommitPhaseError(effect, error);
+          }
+        }
+      }
+      const nextNextEffect = effect.nextEffect;
+      // Remove nextEffect pointer to assist GC
+      effect.nextEffect = null;
+      effect = nextNextEffect;
+    }
+  } else {
```

Comment:
This is the old code path. If we flip the `deferPassiveEffectCleanupDuringUnmount` killswitch off, it will restore the previous behavior.